uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js

```diff
@@ -1104,725 +1116,205 @@ define([
-      });
+        $rootScope.$broadcast('LeaveRequestPopup::ContactSelectionComplete');
+        $rootScope.$digest();
 
-      it('has original opening balance', function () {
-        expect(controller.balance.opening).toBe(expectedOpeningBalance);
+        controller.request.type_id = selectedAbsenceType.id;
       });
 
-      describe('when changing leave type', function () {
-        beforeEach(function () {
-          controller.request.type_id = absenceTypes[1].id;
-          expectedOpeningBalance = absenceTypes[1].remainder;
-
-          $rootScope.$broadcast('LeaveRequestPopup::absenceTypeChanged');
-          $rootScope.$digest();
-        });
-
-        it('uses the opening balance for that leave type', function () {
-          expect(controller.balance.opening).toBe(absenceTypes[1].remainder);
-        });
-
-        describe('when reverting back to the original leave type', function () {
-          it('has original opening balance', function () {
-            expect(controller.balance.opening).toBe(expectedOpeningBalance);
-          });
-        });
+      it('retrieves original balance breakdown', function () {
+        expect(LeaveRequestAPI.getBalanceChangeBreakdown).toHaveBeenCalled();
+        expect(controller.loading.balanceChange).toBe(false);
       });
 
-      describe('when status is admin approved', function () {
-        beforeEach(function () {
-          request = leaveRequestData.all().values[0];
-          request.status_id = getStatusValueFromName(sharedSettings.statusNames.adminApproved);
-          expectedOpeningBalance = absenceTypes[0].remainder - request.balance_change;
-
-          compileComponent({
-            mode: 'edit',
-            request: LeaveRequestInstance.init(request),
-            role: 'manager',
-            selectedAbsenceType: absenceTypes[0]
-          });
-        });
-
-        it('has original opening balance', function () {
-          expect(controller.balance.opening).toBe(expectedOpeningBalance);
-        });
+      it('sets mode to view', function () {
+        expect(controller.isMode('view')).toBeTruthy();
       });
 
-      describe('when status is "Awaiting Approval"', function () {
-        beforeEach(function () {
-          request = leaveRequestData.all().values[0];
-          request.status_id = getStatusValueFromName(sharedSettings.statusNames.awaitingApproval);
-          expectedOpeningBalance = absenceTypes[0].remainder;
-
-          compileComponent({
-            mode: 'edit',
-            request: LeaveRequestInstance.init(request),
-            role: 'manager',
-            selectedAbsenceType: absenceTypes[0]
-          });
-        });
-
-        it('has absence type remainder as opening balance', function () {
-          expect(controller.balance.opening).toBe(expectedOpeningBalance);
-        });
+      it('sets contact id', function () {
+        expect(controller.request.contact_id).toEqual(leaveRequest.contact_id);
```

Comment:
why do we have this test? Isn't this something that comes with the leave request? Should it be:

```js
it('stores the leave request', function () {
  expect(controller.request).toBe(leaveRequest);
});
```

Reply:
It was added long back, but your suggestion is better, changed it.